src/Alluxio.cc

```diff
@@ -18,45 +18,105 @@
 using namespace alluxio;
 using namespace alluxio::jni;
 
-AlluxioClientContext::AlluxioClientContext(const char *host, const char *port)
-{
-    Env env;
-    jvalue  retClientContext;
-    jvalue  retConfiguration;
-    jvalue  retSet;
-    jstring jHostString;
-    jstring jPortString;
-    jobject jHostKeyString;
-    jobject jPortKeyString;
+/**
+   Constructor for AlluxioClientContext
 
-    // First get the ClientContext.Configuration
-    env.callStaticMethod(&retConfiguration, 
-            "alluxio/client/ClientContext", "getConf", "()Lalluxio/Configuration;");
+   This will attach to the JVM thread created for this process.  Prior to
+   calling this AlluxioClientContext::connect() must be called to cause the JVM
+   thread to connect to the master node.
+*/
+AlluxioClientContext::AlluxioClientContext()
+    : m_mustDetachInDtor(false), m_mustDeleteLocalRef(false) {
+  attach();
```

Comment:
Each jni::Env constructor will automatically attach to current thread (through getEnv). The attach call here is redundant? I see that the context needs to detach in destructor. Probably can auto detach in the destructor of jni::Env?